From 404add40962a0cb2b3023e9a1944b5a7f9bba395 Mon Sep 17 00:00:00 2001
From: Bowgo Tsai <bowgotsai@google.com>
Date: Fri, 29 Jun 2018 10:15:34 +0800
Subject: [PATCH 7/8] Add /mnt/product rw mount point for product partitions.

1. init creates /mnt/product used to mount product-specific rw partitions.
2. If a device tree fstab entry for early mount specifies a mount point
   under /mnt/product e.g. /mnt/product/foo, init will create
   /mnt/product/foo mount point.

Bug: 110808288
Test: change dt fstab entry to mount /mnt/product/foo; mount point is
      created correctly, and partition is mounted in early mount.

Change-Id: I321e314992abe1084fd67a382c205f5c0c92bf3d
Merged-In: I321e314992abe1084fd67a382c205f5c0c92bf3d
(cherry picked from commit acc31cee1bd5fe270a22f993fdff79d83f827658)

Former-commit-id: aad498792cd20d0cd9907e5d9525780814fa58be
Former-commit-id: fb452592f1033e7f279859c5a63007d33057fdd3
Former-commit-id: 65fc695e1b8ff7041d9b0f5b4e389a306b6fabfc
---
 init/init.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/init/init.cpp b/init/init.cpp
index 31d79b58c1..93f9576767 100644
--- a/init/init.cpp
+++ b/init/init.cpp
@@ -612,6 +612,9 @@ int main(int argc, char** argv) {
         // /mnt/vendor is used to mount vendor-specific partitions that can not be
         // part of the vendor partition, e.g. because they are mounted read-write.
         mkdir("/mnt/vendor", 0755);
+        // /mnt/product is used to mount product-specific partitions that can not be
+        // part of the product partition, e.g. because they are mounted read-write.
+        mkdir("/mnt/product", 0755);
 
         // Now that tmpfs is mounted on /dev and we have /dev/kmsg, we can actually
         // talk to the outside world...
-- 
2.17.1

